sdk/identity/azure-identity/inc/azure/identity/client_secret_credential.hpp

```diff
@@ -77,7 +70,7 @@ namespace Azure { namespace Identity {
      */
     explicit ClientSecretCredential(
         std::string tenantId,
-        std::string clientId,
+        std::string const& clientId,
```

Comment:
Isn't this breaking?

Reply:
`std::string const& fnParam` <=> `std::string fnParam` is not breaking (either way, vice versa replacement is also not breaking).

Reply:
Really? I thought adding a `const &` to a param is considered breaking. Or does that only apply to adding `const`?

Wouldn't the caller now need to pass in a  `const` local to this function, when previously both const and non-const would have been fine?

Reply:
Adding `const&` to a by-value function param is not breaking. Users can still invoke with the same semantics and pass everything that was been passed before:
```cpp
#include <string>
#include <utility>

void by_cref(std::string const&) {}
void by_val(std::string) {}

static std::string g_s;
std::string& get_ref() { return g_s; }
std::string const& get_cref() { return g_s; }
std::string&& get_rvalref() { return std::move(std::string()); }

int main()
{
    // All the semantics keeps working:
    std::string s;
    by_cref(s);
    by_val(s);

    by_cref({});
    by_val({});

    by_cref("hello");
    by_val("hello");

    by_cref(std::string("hello"));
    by_val(std::string("hello"));

    by_cref(get_ref());
    by_val(get_ref());

    by_cref(get_cref());
    by_val(get_cref());

    by_cref(get_rvalref());
    by_val(get_rvalref());
}
```